src/types.rs

```diff
@@ -992,10 +993,10 @@ impl SourceMap {
         // We want to compare `self` and `adjustment` tokens by line/column numbers in the "original source" file.
         // These line/column numbers are the `dst_line/col` for
         // the `self` tokens and `src_line/col` for the `adjustment` tokens.
-        let self_tokens = std::mem::take(&mut self.tokens);
-        let original_ranges = create_ranges(self_tokens, |t| (t.dst_line, t.dst_col));
-        let adjustment_ranges =
-            create_ranges(adjustment.tokens.clone(), |t| (t.src_line, t.src_col));
+        let mut self_tokens = std::mem::take(&mut self.tokens);
+        let original_ranges = create_ranges(&mut self_tokens, |t| (t.dst_line, t.dst_col));
+        let mut adjustment_tokens = adjustment.tokens.clone();
```

Comment:
I wonder if we could skip this clone in case the tokens are already sorted? Might be another optimization potentially.